### DISCUSSION TITLE
Report Estimated Electricity Use and Carbon Emission

### DISCUSSION THREAD

**Discussion by peterzahemszky** [langchain-ai/langsmith-sdk#794] [Ideas]
### Feature request

Report estimated electricity usage and carbon emission based on public data regarding energy used for training and electricity consumption per input and output token

Possible implementation:
- [x] track input and output token count
- [ ] hard-code approximate electricity intensity of supported models
- [ ] estimate electricity use from input & output token counts
- [ ] convert electricity use to carbon emission using real-time or historical average intensity using existing third-party APIs

Blocked by https://github.com/langchain-ai/langsmith-sdk/issues/756

### Motivation

- Enhance transparency of energy consumption of model training and inference
- Provide insights into the climate impact of model & region choice
- Facilitate more sustainable AI practices by highlighting energy efficiency

**Comment by hinthornw** [comment]
Hi there! This would be cool! I don't think we'll be able to tackle this in a reliable way, however, (at least not any time soon) both because it's hard to get reliable publicly available data and because those numbers are prone to change

**Comment by hinthornw** [comment]
Gonna move to discussions since it's not currently scheduled for work
